lambdas/src/apis/profiles/controllers/profiles.ts

```diff
@@ -11,6 +11,45 @@ import { WearablesOwnership } from '../WearablesOwnership'
 
 const LOGGER = log4js.getLogger('profiles')
 
+function setCacheHeaders(res: Response, profiles: ProfileMetadata[]) {
+  if (profiles.length === 0) return
+
+  let maxTimestamp = profiles[0]?.timestamp
+  for (let i = 1; i < profiles.length; i++) {
+    maxTimestamp = Math.max(maxTimestamp, profiles[i].timestamp)
```

Comment:
You are using this as you can ensure that it was valid when was deployed?

Reply:
Yep. We can assume that if we validated once and it was not updated since, then it is valid now. We don't need to perform the expensive calculations below.